src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js

```diff
@@ -4,88 +4,103 @@ import Flex from "../Flex/Flex";
 import Avatar from "../Avatar/Avatar";
 import ClickableWrapper from "../Clickable/ClickableWrapper";
 import avatarGroupCounterTooltipContentStyles from "./AvatarGroupCounterTooltipContent.module.scss";
+import useEventListener from "../../hooks/useEventListener";
+import { useListenFocusTriggers } from "../../hooks/useListenFocusTriggers";
 
 const KEYS = ["Tab"];
+const ESC = ["Escape"];
 
 export function useTooltipContentTabNavigation({
   counterContainerRef = undefined,
   tooltipContentContainerRef,
   focusPrevPlaceholderRef,
   focusNextPlaceholderRef,
-  setShouldUpdate,
-  setIsTooltipVisible
+  isKeyboardTooltipVisible,
+  setIsKeyboardTooltipVisible
 }) {
-  const hideTooltip = useCallback(() => {
-    // Tricky way to close the tooltip
-    setTimeout(() => {
-      setIsTooltipVisible(false);
-      setIsTooltipVisible(true);
-    });
-  }, [setIsTooltipVisible]);
+  const showKeyboardTooltip = useCallback(() => {
+    if (!isKeyboardTooltipVisible) setIsKeyboardTooltipVisible(true);
+  }, [isKeyboardTooltipVisible, setIsKeyboardTooltipVisible]);
 
-  // For Counter
+  const hideKeyboardTooltip = useCallback(() => {
+    if (isKeyboardTooltipVisible) setIsKeyboardTooltipVisible(false);
+  }, [isKeyboardTooltipVisible, setIsKeyboardTooltipVisible]);
+
+  // Open tooltip manually when keyboard focusing on counter
+  useListenFocusTriggers({
+    ref: counterContainerRef,
+    onFocusByKeyboard: showKeyboardTooltip
+  });
+
+  useEventListener({
+    eventName: "blur",
+    ref: tooltipContentContainerRef,
+    ignoreDocumentFallback: true,
+    callback: hideKeyboardTooltip
+  });
+
+  //Move focus to content by keyboard
   useKeyEvent({
     keys: KEYS,
-    withoutAnyModifier: true,
     ref: counterContainerRef,
+    withoutAnyModifier: true,
     callback: useCallback(
       e => {
-        if (e.target === counterContainerRef.current) {
-          e.preventDefault();
-          tooltipContentContainerRef?.current && tooltipContentContainerRef.current.focus();
-          setShouldUpdate(prev => !prev);
-        }
+        e.preventDefault();
```

Comment:
No condition here, so it's better to put `preventDefault` in the useKeyEvent props